Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/TemplatedProjectExporter.cs

```diff
@@ -19,26 +19,45 @@ public class TemplatedProjectExporter : IProjectExporter
     {
         private readonly DirectoryInfo propsOutputFolder;
 
-        private readonly string solutionFileTemplateText;
-        private readonly string projectFileTemplateText;
-        private readonly string projectPropsFileTemplateText;
-        private readonly string projectTargetsFileTemplateText;
-
-        public TemplatedProjectExporter(DirectoryInfo propsOuputFolder, FileInfo solutionFileTemplatePath, FileInfo projectFileTemplatePath, FileInfo projectPropsFileTemplatePath, FileInfo projectTargetsFileTemplatePath)
+        private readonly FileTemplate projectFileTemplate;
+        private readonly FileTemplate propsFileTemplate;
+        private readonly FileTemplate targetsFileTemplate;
+
+        private readonly FileTemplate solutionFileTemplate;
+
+        /// <summary>
+        /// Creates a new instance of the template driven <see cref="IProjectExporter"/>.
+        /// </summary>
+        /// <param name="propsOutputFolder">The output folder for the projects and props.</param>
+        /// <param name="solutionFileTemplatePath">The path to the solution template.</param>
+        /// <param name="projectFileTemplatePath">The path to the C# project file template.</param>
+        /// <param name="projectPropsFileTemplatePath">The path to the props file template.</param>
+        /// <param name="projectTargetsFileTemplatePath">The path to the targets file template.</param>
+        public TemplatedProjectExporter(DirectoryInfo propsOutputFolder, FileInfo solutionFileTemplatePath, FileInfo projectFileTemplatePath, FileInfo projectPropsFileTemplatePath, FileInfo projectTargetsFileTemplatePath)
         {
-            propsOutputFolder = propsOuputFolder;
+            this.propsOutputFolder = propsOutputFolder;
+
+            FileTemplate.TryParseTemplate(projectFileTemplatePath, out projectFileTemplate);
+            FileTemplate.TryParseTemplate(projectPropsFileTemplatePath, out propsFileTemplate);
+            FileTemplate.TryParseTemplate(projectTargetsFileTemplatePath, out targetsFileTemplate);
 
-            solutionFileTemplateText = File.ReadAllText(solutionFileTemplatePath.FullName);
-            projectFileTemplateText = File.ReadAllText(projectFileTemplatePath.FullName);
-            projectPropsFileTemplateText = File.ReadAllText(projectPropsFileTemplatePath.FullName);
-            projectTargetsFileTemplateText = File.ReadAllText(projectTargetsFileTemplatePath.FullName);
+            FileTemplate.TryParseTemplate(solutionFileTemplatePath, out solutionFileTemplate);
+        }
+
+        ///<inherit-doc/>
```

Comment:
Not sure if this is the same or different to how we use these tags in MRTK, but I believe it's
```suggestion
        /// <inheritdoc />
```